C-edition/src/functions.c

```diff
@@ -124,7 +124,7 @@ void config_load(struct config *config) {
 #ifdef DEBUG
 	debug("configuration file", conf_file);
 #endif
-	file=fopen(conf_file, "r");
+	file=fopen(conf_file, "a+");
```

Comment:
I would rather not use the `a+` flag as it also opens for writing and I don't want to open a file for writing when not needed.

Something like this would be better:
```c
	file = fopen(conf_file, "r");
	if(file==NULL) {
		fprintf(stderr, "Configuration file doesn't exist. attempting to create it.\n");
		file = fopen(conf_file, "a+");
		if(!file) {
			fprintf(stderr, "Failed to create config file '%s': %s\n", conf_file, strerror(errno));
			exit(-1);
		}
	}
```